desktop/main/Wallet.ts

```diff
@@ -160,12 +159,34 @@ const subscribe = () => {
       .catch((error) => ({ error, filePath: null }))
   );
 
-  ipcMain.handle(ipcConsts.W_M_ADD_WALLET_PATH, (_, filePath: string) => {
-    const oldWalletFiles = StoreService.get('walletFiles');
-    const newWalletFiles = R.uniq([...oldWalletFiles, filePath]);
-    StoreService.set('walletFiles', newWalletFiles);
-    return newWalletFiles;
-  });
+  ipcMain.handle(
+    ipcConsts.W_M_ADD_WALLET_PATH,
+    async (
+      _,
+      filePath: string
+    ): Promise<{
+      error: string | null;
+      result: string[] | null;
+    }> => {
```

Comment:
No need to reinvent such types...
```suggestion
    ): Promise<IpcResponse<string[]>> => {
```